src/common/imageio_libraw.c

```diff
@@ -53,32 +53,142 @@ static gboolean _supported_image(const gchar *filename)
 }
 
 
-dt_imageio_retval_t dt_imageio_open_libraw(dt_image_t *img, const char *filename, dt_mipmap_buffer_t *mbuf)
+
+int dt_libraw_lookup_makermodel(const char *maker, const char *model,
+                                   char *mk, int mk_len, char *md, int md_len,
+                                   char *al, int al_len)
 {
-  int err = DT_IMAGEIO_FILE_CORRUPTED;
+  int got_it_done = FALSE;
 
-  if(!_supported_image(filename)) return DT_IMAGEIO_FILE_CORRUPTED;
+  if(g_str_equal(maker, "Canon"))
+  {
+    g_strlcpy(mk, "Canon", mk_len);
 
-  if(!img->exif_inited) (void)dt_exif_read(img, filename);
+    if(g_str_equal(model, "Canon EOS RP"))
+    {
+      g_strlcpy(md, "EOS RP", md_len);
+      g_strlcpy(al, "EOS RP", al_len);
+    }
+    if(g_str_equal(model, "Canon EOS R"))
+    {
+      g_strlcpy(md, "EOS R", md_len);
+      g_strlcpy(al, "EOS R", al_len);
+    }
+    got_it_done = TRUE;
+  }
+
+  return got_it_done;
+}
 
-  libraw_data_t *raw_data = libraw_init(0);
 
-  if(!libraw_open_file(raw_data, filename))
-    goto error;
+dt_imageio_retval_t dt_imageio_open_libraw(dt_image_t *img, const char *filename, dt_mipmap_buffer_t *mbuf)
+{
+  int err;
+  int libraw_err = 0;
+  if(!_supported_image(filename)) return DT_IMAGEIO_FILE_CORRUPTED;
+  if(!img->exif_inited) (void)dt_exif_read(img, filename);
 
+  libraw_data_t *raw = libraw_init(0);
+  if(!raw) return DT_IMAGEIO_FILE_CORRUPTED;
+
+  libraw_err = libraw_open_file(raw, filename);
+  if(libraw_err != LIBRAW_SUCCESS) goto error;
+
+  libraw_err = libraw_unpack(raw);
+  if(libraw_err != LIBRAW_SUCCESS) goto error;
+
+  // Bad method to detect if camera is fully supported by libraw.
+  // But seems to be the best available. libraw crx decoder can actually
+  // decode the raw data, but internal metadata like wb_coeffs, crops etc.
+  // are not populated into libraw structure.
+  if(raw->color.cam_mul[0] == 0.0)
```

Comment:
@TurboGit  The crx decoder in libraw for CR3 can decode afaik every camera file, but if the camera is not explicitly supported, metadata is not properly populated. So we don't get proper WB or crop params which makes processing unusable.
I check the first white balance coeff and if it is zero, metadata is not properly populated.